lib/dal/src/schema/variant.rs

```diff
@@ -968,22 +1072,431 @@ impl SchemaVariant {
 
         let mut auth_funcs = vec![];
 
-        for node_id in workspace_snapshot
-            .outgoing_targets_for_edge_weight_kind(
+        for (edge_weight, _, target_idx) in workspace_snapshot
+            .edges_directed_for_edge_weight_kind(
                 variant_id,
-                EdgeWeightKindDiscriminants::AuthenticationPrototype,
+                Direction::Outgoing,
+                EdgeWeightKindDiscriminants::Func,
             )
             .await?
         {
-            auth_funcs.push(
-                workspace_snapshot
-                    .get_node_weight(node_id)
-                    .await?
-                    .id()
-                    .into(),
-            )
+            if let (EdgeWeightKind::Func(kind), Some(node_weight)) = (
+                edge_weight.kind(),
+                workspace_snapshot.get_node_weight(target_idx).await.ok(),
+            ) {
+                if kind.to_owned() == FuncKind::Authentication {
+                    auth_funcs.push(node_weight.id().into())
+                }
+            }
         }
 
         Ok(auth_funcs)
     }
+    pub async fn list_action_func_ids_for_schema_variant(
```

Comment:
I'd add blank lines between each function definition to adhere to our usual codebase style.